src/jit/lsraxarch.cpp

```diff
@@ -84,47 +95,94 @@ void LinearScan::BuildNode(GenTree* tree)
-            info->srcCount = 0;
-
+        {
+            LclVarDsc* const varDsc = &compiler->lvaTable[tree->AsLclVarCommon()->gtLclNum];
+            if (isCandidateVar(varDsc))
+            {
+                // We handle tracked variables differently from non-tracked ones.  If it is tracked,
+                // we will simply add a use of the tracked variable at its parent/consumer.
+                // Otherwise, for a use we need to actually add the appropriate references for loading
+                // or storing the variable.
+                //
+                // A tracked variable won't actually get used until the appropriate ancestor tree node
+                // is processed, unless this is marked "isLocalDefUse" because it is a stack-based argument
+                // to a call or an orphaned dead node.
+
+                assert(varDsc->lvTracked);
+                unsigned varIndex = varDsc->lvVarIndex;
+
+                // We have only approximate last-use information at this point.  This is because the
+                // execution order doesn't actually reflect the true order in which the localVars
+                // are referenced - but the order of the RefPositions will, so we recompute it after
+                // RefPositions are built.
+                // Use the old value for setting currentLiveVars - note that we do this with the
+                // not-quite-correct setting of lastUse.  However, this is OK because
+                // 1) this is only for preferencing, which doesn't require strict correctness, and
+                // 2) the cases where these out-of-order uses occur should not overlap a kill.
+                // TODO-Throughput: clean this up once we have the execution order correct.  At that point
+                // we can update currentLiveVars at the same place that we create the RefPosition.
+                if ((tree->gtFlags & GTF_VAR_DEATH) != 0)
+                {
+                    VarSetOps::RemoveElemD(compiler, currentLiveVars, varIndex);
+                }
+                INDEBUG(dumpNodeInfo(tree, dstCandidates, 0, 1));
+                return 0;
+            }
+            srcCount = 0;
 #ifdef FEATURE_SIMD
             // Need an additional register to read upper 4 bytes of Vector3.
             if (tree->TypeGet() == TYP_SIMD12)
             {
                 // We need an internal register different from targetReg in which 'tree' produces its result
                 // because both targetReg and internal reg will be in use at the same time.
-                info->internalFloatCount     = 1;
-                info->isInternalRegDelayFree = true;
-                info->setInternalCandidates(this, allSIMDRegs());
+                buildInternalFloatRegisterDefForNode(tree, allSIMDRegs());
+                setInternalRegsDelayFree = true;
+                buildInternalRegisterUses();
             }
 #endif
-            break;
+            RefPosition* def = BuildDef(tree);
```

Comment:
Did you intend to use the `def` variable for anything?

Reply:
No; at one point I had thought I would need to keep the defs around, but realized it was not necessary.